src/Caching/StackExchangeRedis/src/RedisOutputCacheStore.cs

```diff
@@ -0,0 +1,487 @@
+            // for the individual tag sorted-sets, and also the master sorted-set
+            const int EXTEND_EVERY = 250; // some non-trivial number of work
+            int extendCountdown = EXTEND_EVERY;
+            await foreach (var entry in cache.SortedSetScanAsync(_tagMasterKey).WithCancellation(cancellationToken))
+            {
+                await cache.SortedSetRemoveRangeByScoreAsync(GetTagKey((string)entry.Element!), start: 0, stop: keepValuesGreaterThan, flags: GarbageCollectionFlags).ConfigureAwait(false);
+                if (--extendCountdown <= 0)
+                {
+                    await cache.KeyExpireAsync(gcKey, gcLifetime).ConfigureAwait(false);
+                    extendCountdown = EXTEND_EVERY;
+                }
+            }
+            // paying latency on the final master-tag purge: is fine
+            return await cache.SortedSetRemoveRangeByScoreAsync(_tagMasterKey, start: 0, stop: keepValuesGreaterThan).ConfigureAwait(false);
+        }
+        finally
+        {
+            await cache.KeyDeleteAsync(gcKey, CommandFlags.FireAndForget).ConfigureAwait(false);
+        }
+    }
+
+    async ValueTask IOutputCacheStore.EvictByTagAsync(string tag, CancellationToken cancellationToken)
+    {
+        var cache = await ConnectAsync(cancellationToken).ConfigureAwait(false);
+        Debug.Assert(cache is not null);
+
+        // we'll use fire-and-forget on individual deletes, relying on the paging mechanism
+        // of ZSCAN to detect fundamental connection problems - so failure will still be reported
+        const CommandFlags DeleteFlags = CommandFlags.FireAndForget;
+
+        var tagKey = GetTagKey(tag);
+        await foreach (var entry in cache.SortedSetScanAsync(tagKey).WithCancellation(cancellationToken))
+        {
+            await cache.KeyDeleteAsync(GetValueKey((string)entry.Element!), DeleteFlags).ConfigureAwait(false);
+            await cache.SortedSetRemoveAsync(tagKey, entry.Element, DeleteFlags).ConfigureAwait(false);
+        }
+    }
+
+    private RedisKey GetValueKey(string key)
+        => _valueKeyPrefix.Append(key);
+
+    private RedisKey GetTagKey(string tag)
+        => _tagKeyPrefix.Append(tag);
+
+    async ValueTask<byte[]?> IOutputCacheStore.GetAsync(string key, CancellationToken cancellationToken)
+    {
+        ArgumentNullThrowHelper.ThrowIfNull(key);
+
+        var cache = await ConnectAsync(cancellationToken).ConfigureAwait(false);
+        Debug.Assert(cache is not null);
+
+        try
+        {
+            return (byte[]?)(await cache.StringGetAsync(GetValueKey(key)).ConfigureAwait(false));
+        }
+        catch (Exception ex)
+        {
+            OnRedisError(ex, cache);
+            throw;
+        }
+    }
+
+    async ValueTask<bool> IOutputCacheBufferWriterStore.GetAsync(string key, IBufferWriter<byte> destination, CancellationToken cancellationToken)
```

Comment:
Nit: should use the `private`-modifyer explicitly.

Reply:
it is an explicit interface implementation; accessibility modifiers are not valid

Reply:
Oh, yes -- overlooked that, thanks!